substreams/ethereum-uniswap-v4/src/modules/1_map_pool_created.rs

```diff
@@ -0,0 +1,124 @@
+use std::str::FromStr;
+
+use ethabi::ethereum_types::Address;
+use substreams::scalar::BigInt;
+use substreams_ethereum::pb::eth::v2::{self as eth};
+
+use substreams_helper::{event_handler::EventHandler, hex::Hexable};
+
+use crate::abi::pool_manager::events::Initialize;
+
+use tycho_substreams::prelude::*;
+#[substreams::handlers::map]
+pub fn map_pools_created(
+    params: String,
+    block: eth::Block,
+) -> Result<BlockEntityChanges, substreams::errors::Error> {
+    let mut new_pools: Vec<TransactionEntityChanges> = vec![];
+    let pool_manager = params.as_str();
+
+    get_new_pools(&block, &mut new_pools, pool_manager);
+
+    Ok(BlockEntityChanges { block: None, changes: new_pools })
+}
+
+// Extract new pools initialized on the pool manager contract
+fn get_new_pools(
+    block: &eth::Block,
+    new_pools: &mut Vec<TransactionEntityChanges>,
+    pool_manager_address: &str,
+) {
+    // Extract new pools from Initialize events
+    let mut on_pool_created = |event: Initialize, _tx: &eth::TransactionTrace, _log: &eth::Log| {
+        let tycho_tx: Transaction = _tx.into();
+
+        new_pools.push(TransactionEntityChanges {
+            tx: Some(tycho_tx.clone()),
+            entity_changes: vec![EntityChanges {
+                component_id: event.id.to_vec().to_hex(),
+                attributes: vec![
+                    Attribute {
+                        name: "liquidity".to_string(),
+                        value: BigInt::from(0).to_signed_bytes_be(),
+                        change: ChangeType::Creation.into(),
+                    },
+                    Attribute {
+                        name: "tick".to_string(),
+                        value: event.tick.to_signed_bytes_be(),
+                        change: ChangeType::Creation.into(),
+                    },
+                    Attribute {
+                        name: "protocol_fees/zero2one".to_string(),
```

Comment:
Do I need to initialize this as 0? We didn't do it for UniV3 - but I think it's weird to Update something that wasn't created before - I'm not sure we enforce it - if we do, we might have a bug on Univ3

Reply:
Nope we don't enforce it, we added the default because when parsing pool state it's better to have a consistent set of attributes than omit some attributes because they were never updated. 

Agreed that we should have it on Uniswap v3 though. 
Protocol fees isn't really a thing there so it was overlooked because nobody cares about it.